_examples/fruits_test.go

```diff
@@ -59,10 +65,31 @@ func TestFruits(t *testing.T) {
 	obj.Value("colors").Array().Elements("green", "red")
 	obj.Value("colors").Array().Element(0).String().Equal("green")
 	obj.Value("colors").Array().Element(1).String().Equal("red")
+	obj.Value("colors").Array().Element(1).String().IsASCII()
+	obj.Value("colors").Array().Element(1).String().HasPrefix("re")
+	obj.Value("colors").Array().Element(1).String().HasSuffix("ed")
 
 	obj.Value("weight").Number().Equal(200)
 
 	e.GET("/fruits/melon").
 		Expect().
 		Status(http.StatusNotFound)
+
+	var fruits fruitMapList
+	apple["type"] = "fruit"
+	orange["type"] = "fruit"
+	fruits = append(fruits, apple)
+	fruits = append(fruits, orange)
+
+	fruitsData := e.GET("/fruits/with-data").WithJSON(fruits).
+		Expect().
+		Status(http.StatusOK).JSON().Array()
```

Comment:
The original fruits server has get/put logic: you put some data and then you can get it.

The new endpoint instead has echo server logic: you send data in body and receive it back.

To keep example simpler, let's use the same logic everywhere. Instead of adding echo server, let's put all necessary data at the beginning of the test, as we already do.

I think we don't need to add new endpoint at all, to get a map. We already have one: `/fruit/<xxx>` will return a map and we can put whatever map we want. So let's just reuse it and put a more complicated fruit and inspect it here.

Reply:
But we can put array of maps into one of the fruits, can't we?

Reply:
ah ok I see you're right will do that